### PR TITLE
Deprecate `--new-formula`/`--new-cask` options.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

See https://github.com/Homebrew/homebrew-cask/pull/161724.

Both `--new-formula` as well as `--new-cask` do the same thing as `--new`, so they're not needed anymore.

### CODE REVIEW

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -44,10 +44,20 @@ def self.audit_args
...
+      switch "--new",
              description: "Run various additional style checks to determine if a new formula or cask is eligible " \
-                          "for Homebrew. This should be used when creating new formula and implies " \
+                          "for Homebrew. This should be used when creating new formulae or casks and implies " \
                           "`--strict` and `--online`."
+      switch "--new-formula",
+             description: "Run various additional style checks to determine if a new formula is eligible " \
+                          "for Homebrew. This should be used when creating new formulae and implies " \
+                          "`--strict` and `--online`.",
+             replacement: "--new"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb:55]:
```suggestion
             replacement: "--new",
             hidden: true 
```

File: Library/Homebrew/dev-cmd/audit.rb [hunk 2]
```diff
@@ -44,10 +44,20 @@ def self.audit_args
...
+      switch "--new-cask",
+             description: "Run various additional style checks to determine if a new cask is eligible " \
+                          "for Homebrew. This should be used when creating new casks and implies " \
+                          "`--strict` and `--online`.",
+             replacement: "--new"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb:60]:
```suggestion
             replacement: "--new",
             hidden: true 
```

so we can treat this as deprecated

File: Library/Homebrew/dev-cmd/audit.rb [hunk 3]
```diff
@@ -101,14 +111,16 @@ def self.audit_args
   def self.audit
     args = audit_args.parse
 
+    new_cask = args.new? || args.new_cask?
+    new_formula = args.new? || args.new_formula?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb:115]:
```suggestion
    new_cask = args.new?
    if args.new_cask?
      # odeprecated "brew audit --new-cask"
      new_cask = true
    end
    new_formula = args.new?
    if args.new_formula?
      # odeprecated "brew audit --new-formula"
      new_formula = true
    end
```

**Review by MikeMcQuaid — COMMENTED** [review]
Thanks @reitermarkus, looking good so far. A few suggestions so we can get this deprecated sooner rather than later.

File: Library/Homebrew/dev-cmd/audit.rb (hunk 2 above)
Comment (reitermarkus) [reply Library/Homebrew/dev-cmd/audit.rb:60]: Currently, specifying `replacement` implies it is already `odeprecated` inside `parser.rb`. I guess we just make it `hidden` without `replacement`.

File: Library/Homebrew/dev-cmd/audit.rb (hunk 3 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/audit.rb:113]: @reitermarkus Still think this would be good to do so we delete and print deprecation messages on usage.

Comment (reitermarkus) [reply Library/Homebrew/dev-cmd/audit.rb:113]: `odeprecated` is automatically done by `CLI::Parser` once the `replacement`s are uncommented on the `switch` calls.

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/audit.rb:113]: Yup but we're reliant on the commenting and changing of `odeprecated` to `odisabled` process to get this code removed.

Comment (reitermarkus) [reply Library/Homebrew/dev-cmd/audit.rb:117]: I already added a `odeprecated` comment above on the `switch` calls. Once the `switch` calls use `disable: true`, calling `args.new_formula?`/`args.new_cask?` here will fail CI anyways.

Comment (reitermarkus) [reply Library/Homebrew/dev-cmd/audit.rb:117]: Also added a comment and `disable: false` to make it greppable.

**Review by MikeMcQuaid — APPROVED** [review]
Thanks again @reitermarkus!

**Pull request merged**